tests/unit/api/test_feature_list.py

```diff
@@ -767,8 +760,7 @@ def test_feature_list_update_status_and_default_version_mode__unsaved_feature_li
     assert feature_list.saved is False
     with pytest.raises(RecordRetrievalException) as exc:
         feature_list.update_status(FeatureListStatus.TEMPLATE)
-    namespace_id = feature_list.feature_list_namespace_id
-    expected = f'FeatureListNamespace (id: "{namespace_id}") not found. Please save the FeatureList object first.'
+    expected = f'FeatureList (id: "{feature_list.id}") not found. Please save the FeatureList object first.'
```

Comment:
Curious how do the changes cause this error message to change? The new error message does look better.

Reply:
Previously, we store the `feature_list_namespace_id` at the `FeatureList` class and use it to retrieve the corresponding feature list namespace. With the refactoring, we did not store the `feature_list_namespace_id`. We retrieve it by accessing the retrieved feature list. If the feature list is not saved, it will throw the missing feature list error first (rather than missing feature list namespace error).